std/math.d

```diff
@@ -7449,6 +7449,9 @@ real fma(real x, real y, real z) @safe pure nothrow @nogc { return (x * y) + z;
 Unqual!F pow(F, G)(F x, G n) @nogc @trusted pure nothrow
 if (isFloatingPoint!(F) && isIntegral!(G))
 {
+    // NaN ^^ 0 is an exception defined by IEEE
+    if (isNaN(x) && n != 0) return x;
```

Comment:
```suggestion
    if (isNaN(x)) return n ? x : 1.0;
```

should help with the slowness of nan

Reply:
At least on my box this had no effect...